src/tables.jl

```diff
@@ -0,0 +1,100 @@
+const AG = ArchGDAL
+
+struct GeoTable{T<:Union{IFeatureLayer, FeatureLayer}} 
+    layer::T
+end
+
+function geotable(layer)
+    featuredefn = layerdefn(layer)
+    ngeometries = ngeom(featuredefn)
+    GeoTable(layer)
+end
+
+function Tables.schema(layer::AG.AbstractFeatureLayer)
+    nfield = AG.nfield(layer)
+    featuredefn = AG.layerdefn(layer)
+    ngeom = ArchGDAL.ngeom(featuredefn)
+
+    fielddefns = (AG.getfielddefn(featuredefn, i) for i in 0:nfield-1)
+    names_fields = Tuple(AG.getname(fielddefn) for fielddefn in fielddefns)
+    types_fields = Tuple(AG._FIELDTYPE[AG.gettype(fielddefn)] for fielddefn in fielddefns)
+    Tables.Schema(names_fields, types_fields)
+end
+
+function Base.iterate(gt::GeoTable, st = 0)
+    layer = gt.layer
+    AG.resetreading!(layer)
+    nfeat = AG.nfeature(layer)
+    nfield = AG.nfield(layer)
+    ngeom = AG.ngeom(layer)
+    featuredefn = AG.layerdefn(layer)
+
+    name = []
+    v = []
+    for field_no in 0:nfield-1
+        field = AG.getfielddefn(featuredefn, field_no)
+        push!(name, AG.getname(field))
+    end
+    push!(name, "geometry")
+    
+    st >= nfeat && return nothing
+    AG.getfeature(layer, st) do feature
```

Comment:
So the main reason of using `getfeature` instead of `nextfeature` here is that we can link the row/feature number directly with the value of `st`. This is useful when we are calling the `iterate` method iteratively eg: while constructing a `DataFrame` .
Now the issue with `getfeature` arises when we have two different file formats, eg: GeoJSON starts from 0, while geopackage layers have features starting from 1, so if st is initially 0, geojson works while geopackage throws a Null Feature hence an error. 

Now this is handled well by `nextfeature`, where we don't have to worry about the indices. But the thing is again if we call the iterate method repeatedly(eg DataFrame.),for the first time it works well, but now at the end the layer reading is +1 the size of the layer. So if we call the DataFrame() method again, it starts with a NullFeature and hence an error. 

Apparently `AG.resetreading!(layer)` is made to reset the layer's feature number reading. But since Base.iterate is called repeatedly, using it will keep returning the first feature only.

Reply:
Here are a few possible workarounds:
* We use `getfeature` and apply an if condition in the beginning to check for indexing and direct it appropriately.
* or perhaps use the `nextfeature` method and specify in the docs those consuming the API must use `AG.resetreading!(layer)` at after every iterative call?

Reply:
What, I am really surprised if `getfeature` starts from 0 in format A but 1 in format B??

But I think `nextfeature` would be better anyway. Isn't it enough to do `if iszero(st) resetreading`?